### ISSUE TITLE
Add support for  SMBv2 and SMBv3 and not only SMBv1

### ISSUE THREAD

**Issue by courville** [nova-video-player/aos-AVP#2]
Since most of the OS are deactivating SMBv1, switching from [jcifs](https://jcifs.samba.org/) to [jcifs-ng](https://github.com/AgNO3/jcifs-ng) is the identified way to go.
There is some work started in trying to get SMB2+ (not yet functional) here: https://github.com/nnoury in devSmb branches.

**Comment by courville** [comment]
This includes reworking of network share discovery process.

**Comment by courville** [comment]
jcifsng branches for Video FileCoreLibrary and MediaLib from nnoury consist in an WIP implementation. It works with win10 and synology but not yet with macos. More testing needs to be done before release.
https://github.com/nova-video-player/aos-FileCoreLibrary/commits/jcifsng
https://github.com/nova-video-player/aos-MediaLib/commits/jcifsng
https://github.com/nova-video-player/aos-Video/commits/jcifsng

**Comment by courville** [comment]
Experimentations merging github/jcifsng onto current branch leads to:
- synology SMBv1 discovery and login works
- synology SMBv2 only mode discovery and login works
- when accessing a root level protected share attempts to login as guest with sandbox never stopping (i.e. not requesting credentials)
- regression in browsing of behavior: only directories for which the authenticated user has permission to access are listed. Before all were listed and if a permission error was encountered, credentials were asked again

**Comment by courville** [comment]
For now with this program https://gist.github.com/courville/a0e6fe1ce2f31c9adc52191216eed3e0 I fail to list server shares when it is password protected with SMB1.

**Comment by courville** [comment]
Is it possible to browse shares of a server?
https://github.com/AgNO3/jcifs-ng/issues/22

**Comment by courville** [comment]
Reporting current issue to jcifs-ng author https://github.com/AgNO3/jcifs-ng/issues/136

**Comment by courville** [comment]
> Reporting current issue to jcifs-ng author [AgNO3/jcifs-ng#136](https://github.com/AgNO3/jcifs-ng/issues/136)

Issue fixed on latest jcifs-ng! Cool.

**Comment by courville** [comment]
Now status is that browsing on root and shares of SMB2 server works on synology and win10.

**Comment by courville** [comment]
SMB2 root browsing on a MacOSX server is KO for now https://github.com/AgNO3/jcifs-ng/issues/138
However adding manually URI a MacOSX SMB network share works.

**Comment by courville** [comment]
Another issue is met on Swisscom Internet Box 2 when SMB2 is enabled on jcifs-ng (OK with SMB1 only): https://github.com/AgNO3/jcifs-ng/issues/137

**Comment by courville** [comment]
SMB2 browsing on macOSX fixed https://github.com/nova-video-player/aos-FileCoreLibrary/commit/0be960c42f6fed30df8196ac39189adf8e87becd

**Comment by courville** [comment]
Apparently win7 SMB1 browsing on non protected shares requests login/password :-(
To be investigated.
Here is the scarce report: https://github.com/nova-video-player/aos-AVP/issues/50
I have reproduced the issue on winxp too cf. https://github.com/AgNO3/jcifs-ng/issues/148

**Comment by courville** [comment]
On HP printer without usb key you have a crash with:
```
03-10 14:41:53.974  6751  7135 E AndroidRuntime: FATAL EXCEPTION: Thread-32
03-10 14:41:53.974  6751  7135 E AndroidRuntime: Process: org.courville.nova, PID: 6751
03-10 14:41:53.974  6751  7135 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.connectHost(SmbTreeConnection.java:504)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.connectHost(SmbTreeConnection.java:484)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.connect(SmbTreeConnection.java:460)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.connectWrapException(SmbTreeConnection.java:421)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.resolveDfs0(SmbTreeConnection.java:684)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.ensureDFSResolved(SmbTreeConnection.java:653)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:393)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:314)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:294)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbEnumerationUtil.doNetShareEnum(SmbEnumerationUtil.java:104)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbEnumerationUtil.doShareEnum(SmbEnumerationUtil.java:160)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbEnumerationUtil.doEnum(SmbEnumerationUtil.java:222)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbEnumerationUtil.listFiles(SmbEnumerationUtil.java:283)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at jcifs.smb.SmbFile.listFiles(SmbFile.java:1266)
03-10 14:41:53.974  6751  7135 E AndroidRuntime: 	at com.archos.filecorelibrary.jcifs.JcifListingEngine$JcifListingThread.run(JcifListingEngine.java:122)
```
Specific issue is tracked here: https://github.com/AgNO3/jcifs-ng/issues/149

**Comment by courville** [comment]
All issues fixed for now.

**Comment by courville** [comment]
TODO: check if the netbios/IP address caching performed in the discovery process is still active.
It seems not since jcifs-old works with netbios name different than DNS one and not anymore ("jacqueline" computer issue) with jcifs-ng.
Check if any former resolvOrder modifications using caching are absent.

**Comment by courville** [comment]
We have separated the discovery using pieces of jcifs-old in src/jcifs2.
This implements caching via in memory hash table in src/jcifs2/netbios/Lmhosts.java that is not accessible from jcifs-ng.
This causes regression on browsing speed and access in emulator/chromebook since we need to rediscover at each time netbios/IP using BCAST with long timeouts.
Expose the need and limitation to jcifs-ng author.

**Comment by courville** [comment]
In the past we hacked the src/jcifs2/netbios/Lmhosts.java to insert a hash table that the out of band discovery process could feed inserting a dependency to com.archos.filecorelibrary.samba.SambaDiscovery..
We understand that there is a cache mechanism into jcifs-ng in src/main/java/jcifs/netbios/NameServiceClientImpl.java.
Would there be a public API to inject netbios/IP out of band discoveries into the cache.
Better would be to reinject the changes we made in the out of band discovery process into jcifs to fully rely on jcifs-ng new APIs.
Note that the caching mechanism is a -ng addition and was not present in the -old jcifs.

**Comment by courville** [comment]
Mind the "Samba Discovery Nat traversal for ChromeOS Android containment" and getDoubleNatIpAddress() that is present in  src/com/archos/filecorelibrary/samba that helps for Chromebooks and also emulator. Taking this in for jcifs-ng might be difficult...
Perhaps we need our own resolver?

**Comment by Pentaphon** [comment]
Will this make it to v4?

**Comment by courville** [comment]
> Will this make it to v4?

I am afraid not but active development and stabilization restarted in jcifs-ng branches (not jcifs).

**Comment by courville** [comment]
One crash found by @nnoury 
```
09-13 15:47:39.270 11308 11308 D IndexHelper: writeVideoInfo true
09-13 15:47:39.317 11308 15055 W System.err: jcifs.smb.SmbException: Failed to acquire credits in time
09-13 15:47:39.318 11308 15055 W System.err:        at jcifs.smb.SmbTransportImpl.sendrecv(SmbTransportImpl.java:980)
09-13 15:47:39.318 11308 15055 W System.err:        at jcifs.smb.SmbTransportImpl.send(SmbTransportImpl.java:1543)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbSessionImpl.send(SmbSessionImpl.java:409)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:472)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:404)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:318)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:298)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
09-13 15:47:39.319 11308 15055 W System.err:        at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:117)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.DirFileEntryEnumIterator2.open(DirFileEntryEnumIterator2.java:94)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.DirFileEntryEnumIteratorBase.<init>(DirFileEntryEnumIteratorBase.java:67)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.DirFileEntryEnumIterator2.<init>(DirFileEntryEnumIterator2.java:56)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.SmbEnumerationUtil.doEnum(SmbEnumerationUtil.java:223)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.SmbEnumerationUtil.listFiles(SmbEnumerationUtil.java:279)
09-13 15:47:39.320 11308 15055 W System.err:        at jcifs.smb.SmbFile.listFiles(SmbFile.java:1199)
09-13 15:47:39.321 11308 15055 W System.err:        at com.archos.filecorelibrary.jcifs.JcifsRawLister.getFileList(JcifsRawLister.java:58)
09-13 15:47:39.321 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb.getListOfDBForUri(XmlDb.java:327)
09-13 15:47:39.321 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb.deleteAssociatedResumeDatabase(XmlDb.java:439)
09-13 15:47:39.321 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb.writeXml(XmlDb.java:464)
09-13 15:47:39.321 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb.access$300(XmlDb.java:59)
09-13 15:47:39.322 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb$WriteTask.doInBackground(XmlDb.java:228)
09-13 15:47:39.322 11308 15055 W System.err:        at com.archos.mediacenter.utils.videodb.XmlDb$WriteTask.doInBackground(XmlDb.java:214)
09-13 15:47:39.322 11308 15055 W System.err:        at android.os.AsyncTask$2.call(AsyncTask.java:333)
09-13 15:47:39.322 11308 15055 W System.err:        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-13 15:47:39.322 11308 15055 W System.err:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
09-13 15:47:39.322 11308 15055 W System.err:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
09-13 15:47:39.322 11308 15055 W System.err:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
09-13 15:47:39.323 11308 15055 W System.err:        at java.lang.Thread.run(Thread.java:764)
```

**Comment by courville** [comment]
https://github.com/nova-video-player/aos-Video/commit/396db5c6b41cfcf28f788c0e1453ae78a3cfc95a is a workaround to address crash while seeking:
```
09-14 16:39:28.942 17187 17211 D avos_player: 0x7019423200|avos_mp_create:
09-14 16:39:28.967 17187 17187 W System.err: android.os.NetworkOnMainThreadException
09-14 16:39:28.967 17187 17187 W System.err: 	at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1450)
09-14 16:39:28.967 17187 17187 W System.err: 	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:108)
09-14 16:39:28.968 17187 17187 W System.err: 	at java.net.SocketOutputStream.write(SocketOutputStream.java:153)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbTransportImpl.doSend(SmbTransportImpl.java:914)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.util.transport.Transport.doSend(Transport.java:276)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.util.transport.Transport.sendrecv(Transport.java:215)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbTransportImpl.sendrecv(SmbTransportImpl.java:1025)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbTransportImpl.send(SmbTransportImpl.java:1543)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbSessionImpl.send(SmbSessionImpl.java:409)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbTreeImpl.send(SmbTreeImpl.java:472)
09-14 16:39:28.968 17187 17187 W System.err: 	at jcifs.smb.SmbTreeConnection.send0(SmbTreeConnection.java:404)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:318)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbTreeConnection.send(SmbTreeConnection.java:298)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:130)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbTreeHandleImpl.send(SmbTreeHandleImpl.java:117)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbFile.withOpen(SmbFile.java:1741)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbFile.withOpen(SmbFile.java:1710)
09-14 16:39:28.969 17187 17187 W System.err: 	at jcifs.smb.SmbFile.withOpen(SmbFile.java:1704)
09-14 16:39:28.970 17187 17187 W System.err: 	at jcifs.smb.SmbFile.queryPath(SmbFile.java:770)
09-14 16:39:28.970 17187 17187 W System.err: 	at jcifs.smb.SmbFile.exists(SmbFile.java:851)
09-14 16:39:28.970 17187 17187 W System.err: 	at jcifs.smb.SmbFile.isDirectory(SmbFile.java:1063)
09-14 16:39:28.970 17187 17187 W System.err: 	at com.archos.filecorelibrary.jcifs.JcifsFile2.<init>(JcifsFile2.java:109)
09-14 16:39:28.970 17187 17187 W System.err: 	at com.archos.filecorelibrary.jcifs.JcifsFile2.fromUri(JcifsFile2.java:135)
09-14 16:39:28.970 17187 17187 W System.err: 	at com.archos.filecorelibrary.MetaFile2Factory.getMetaFileForUrl(MetaFile2Factory.java:33)
09-14 16:39:28.970 17187 17187 W System.err: 	at com.archos.medialib.SmbProxy.start(SmbProxy.java:45)
09-14 16:39:28.970 17187 17187 W System.err: 	at com.archos.medialib.SmbProxy.setDataSource(SmbProxy.java:68)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.medialib.AvosMediaPlayer.setDataSource(AvosMediaPlayer.java:99)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.medialib.AvosMediaPlayer.setDataSource(AvosMediaPlayer.java:142)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.Player.openVideo(Player.java:449)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.Player.setVideoURI(Player.java:372)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.PlayerService.onStreamingUriOK(PlayerService.java:639)
09-14 16:39:28.971 17187 17192 I zygote64: Do full code cache collection, code=250KB, data=226KB
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.PlayerService.onDataUriOK(PlayerService.java:598)
09-14 16:39:28.971 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.PlayerService.setVideoInfo(PlayerService.java:1005)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.PlayerActivity$PlayerListener.onVideoDb(PlayerActivity.java:3503)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.video.player.PlayerService.onVideoDb(PlayerService.java:1034)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.utils.videodb.IndexHelper.onVideoDbInfo(IndexHelper.java:339)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.utils.videodb.IndexHelper.access$100(IndexHelper.java:49)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.utils.videodb.IndexHelper$XmlObserver.onParseOk(IndexHelper.java:204)
09-14 16:39:28.972 17187 17187 W System.err: 	at com.archos.mediacenter.utils.videodb.XmlDb.handleMessage(XmlDb.java:598)
09-14 16:39:28.972 17187 17187 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:102)
09-14 16:39:28.973 17187 17187 W System.err: 	at android.os.Looper.loop(Looper.java:164)
09-14 16:39:28.973 17187 17187 W System.err: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
09-14 16:39:28.973 17187 17187 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
09-14 16:39:28.973 17187 17192 I zygote64: After code cache collection, code=246KB, data=149KB
09-14 16:39:28.973 17187 17187 W System.err: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
09-14 16:39:28.973 17187 17187 W System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
09-14 16:39:28.979 17187 17211 D avos_player: 0x7019423200|avos_mp_setdatasource: http://localhost:40386/Chuck.S05E01.HDTV.XviD-LOL.avi
09-14 16:39:28.979 17187 17187 D avos_media_player: setVideoSurface
09-14 16:39:28.979 17187 17187 V avos_media_player: free_native_window
```

**Comment by courville** [comment]
Remove crash returning from floating player: https://github.com/nova-video-player/aos-Video/commit/393d8a9a2930d00e69ee7257822531461b0dd024

**Comment by courville** [comment]
Crash seen when browsing non indexed video shares and opening a tvshow episode, it gets indexed and boum:
```
E/JcifsFileEditor( 8419): SmbException or MalformedURLException:
E/JcifsFileEditor( 8419): jcifs.smb.SmbException: The network name cannot be found.
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTransportImpl.checkStatus2(SourceFile:1461)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTransportImpl.checkStatus(SourceFile:1572)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTransportImpl.sendrecv(SourceFile:1027)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTransportImpl.send(SourceFile:1543)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbSessionImpl.send(SourceFile:409)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbSessionImpl.send(SourceFile:347)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeImpl.treeConnect(SourceFile:611)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeConnection.connectTree(SourceFile:614)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeConnection.connectHost(SourceFile:568)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeConnection.connectHost(SourceFile:489)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeConnection.connect(SourceFile:465)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbTreeConnection.connectWrapException(SourceFile:426)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbFile.ensureTreeConnected(SourceFile:551)
E/JcifsFileEditor( 8419):       at jcifs.smb.SmbFile.exists(SourceFile:845)
E/JcifsFileEditor( 8419):       at com.archos.filecorelibrary.jcifs.JcifsFileEditor.exists(SourceFile:131)
E/JcifsFileEditor( 8419):       at com.archos.mediaprovider.video.VideoStoreImportImpl.isNoMediaPath(SourceFile:609)
E/JcifsFileEditor( 8419):       at com.archos.mediacenter.video.leanback.details.VideoDetailsFragment$9.run(SourceFile:869)
```

**Comment by courville** [comment]
Crash after indexing on AndroidTV, main view not updated, going into network shares and boum:
```
--------- beginning of crash
09-18 23:07:44.105  9105  9153 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
09-18 23:07:44.105  9105  9153 E AndroidRuntime: Process: org.courville.nova, PID: 9105
09-18 23:07:44.105  9105  9153 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at android.os.AsyncTask$3.done(AsyncTask.java:353)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:764)
09-18 23:07:44.105  9105  9153 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.File android.content.Conte
xt.getDatabasePath(java.lang.String)' on a null object reference
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:292)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:238)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at com.archos.mediacenter.video.browser.ShortcutDb.getCursorAllShortcuts(SourceFile:102)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at com.archos.mediacenter.video.leanback.network.NetworkRootFragment$FtpShortcutsLoaderTask.doInBackgrou
nd(SourceFile:382)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at com.archos.mediacenter.video.leanback.network.NetworkRootFragment$FtpShortcutsLoaderTask.doInBackground(SourceFile:379)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at android.os.AsyncTask$2.call(AsyncTask.java:333)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-18 23:07:44.105  9105  9153 E AndroidRuntime:        ... 4 more
```

Seems not really realated to jcifs-ng

**Comment by Pentaphon** [comment]
I tried the 5.12 SMBv2/3 experimental release today I am unable to connect to unpassworded Windows 10 64 bit SMB shares while I can successfully connect to them with Kodi 18.6

**Comment by Pentaphon** [comment]
@courville probably should close this issue.

**Comment by courville** [comment]
@Pentaphon let's wait a little more that the deployment goes ok and there is not again any rollback... cf. https://github.com/nova-video-player/aos-AVP/issues/398

**Comment by courville** [comment]
Current attempt of deploying jcifs-ng in v5.15.x failed: many connectivity problems were spotted and reported with more or less description https://www.reddit.com/r/NovaVideoPlayer/comments/k793s6/not_reading_smb_on_firestick/ and https://www.reddit.com/r/NovaVideoPlayer/comments/k79t6u/smb_problem_please_make_sure_you_upgraded_to/
The version of jcifs-ng library used was https://github.com/AgNO3/jcifs-ng/commit/b5ad20d5b80caabeef76ca5f84a0379eb1d82b0f
Some issues were addressed:
- changing jcifs resolveOrder makes connection issue due to wrong server IP used https://github.com/nova-video-player/aos-AVP/issues/398 (fixed) https://github.com/AgNO3/jcifs-ng/issues/258
- trying to cope with WD NAS connectivity issue but without success https://github.com/nova-video-player/aos-FileCoreLibrary/commit/271d69921f161e93128e46fe76844ec541ea160e i.e. https://github.com/AgNO3/jcifs-ng/issues/226 is still open
Still some issues are open to tackle:
- SMB not working if no Internet on local network  https://github.com/nova-video-player/aos-AVP/issues/403
- I can reproduce jcifs-ng endless sandglass when opening a remote share the first time https://github.com/nova-video-player/aos-AVP/issues/377 still and might be related to https://github.com/AgNO3/jcifs-ng/issues/214

Current magic options used are https://github.com/AgNO3/jcifs-ng/issues/259 i.e. 
```
if (isSmb2) {
	prop.put("jcifs.smb.client.disableSMB1", "true");
	prop.put("jcifs.smb.client.enableSMB2", "true");
	// note that connectivity with smbV1 will not be working
	prop.put("jcifs.smb.client.useSMB2Negotiation", "true");
	// disable dfs makes win10 shares with ms account work
	prop.put("jcifs.smb.client.dfs.disabled", "true");
} else {
	prop.put("jcifs.smb.client.disableSMB1", "false");
	prop.put("jcifs.smb.client.enableSMB2", "false");
	prop.put("jcifs.smb.client.useSMB2Negotiation", "false");
	// see https://github.com/AgNO3/jcifs-ng/issues/226
	prop.put("jcifs.smb.useRawNTLM", "true");
}

// get around https://github.com/AgNO3/jcifs-ng/issues/40 and this is required for guest login on win10 smb2
prop.put("jcifs.smb.client.ipcSigningEnforced", "false");
// allow plaintext password fallback
prop.put("jcifs.smb.client.disablePlainTextPasswords", "false");

// Required to make WD MyCloud work cf. https://github.com/AgNO3/jcifs-ng/issues/225
// made guest work on Win10 https://github.com/AgNO3/jcifs-ng/issues/186
prop.put("jcifs.smb.client.disableSpnegoIntegrity", "true");
```

**Comment by courville** [comment]
v5.x and v6.x has SMBv2 support integrated.
